tree/ntuple/v7/src/RPageStorageFile.cxx

```diff
@@ -85,19 +85,13 @@ ROOT::Experimental::Detail::RPageSinkFile::~RPageSinkFile()
 {
 }
 
-
-void ROOT::Experimental::Detail::RPageSinkFile::CreateImpl(const RNTupleModel & /* model */)
+void ROOT::Experimental::Detail::RPageSinkFile::CreateImpl(const RNTupleModel & /* model */,
+                                                           unsigned char *serializedHeader, std::uint32_t length)
 {
-   const auto &descriptor = fDescriptorBuilder.GetDescriptor();
-   fSerializationContext = Internal::RNTupleSerializer::SerializeHeaderV1(nullptr, descriptor);
-   auto buffer = std::make_unique<unsigned char []>(fSerializationContext.GetHeaderSize());
-   fSerializationContext = Internal::RNTupleSerializer::SerializeHeaderV1(buffer.get(), descriptor);
-
-   auto zipBuffer = std::make_unique<unsigned char[]>(fSerializationContext.GetHeaderSize());
-   auto szZipHeader =
-      fCompressor->Zip(buffer.get(), fSerializationContext.GetHeaderSize(), GetWriteOptions().GetCompression(),
-         [&zipBuffer](const void *b, size_t n, size_t o){ memcpy(zipBuffer.get() + o, b, n); } );
-   fWriter->WriteNTupleHeader(zipBuffer.get(), szZipHeader, fSerializationContext.GetHeaderSize());
+   auto zipBuffer = std::make_unique<unsigned char[]>(length);
```

Comment:
Might the output data be larger than the allocated buffer?  I guess not (due to https://github.com/root-project/root/blob/2b5ea5e9c1555696fdb1d392651864f7e64a90e1/tree/ntuple/v7/inc/ROOT/RNTupleZip.hxx#L84).

Reply:
Correct